cmd/thor/node/node.go

```diff
@@ -396,6 +396,7 @@ func (n *Node) processBlock(newBlock *block.Block, stats *blockStats) (bool, err
 		metricBlockProcessedTxs().SetWithLabel(int64(len(receipts)), map[string]string{"type": "received"})
 		metricBlockProcessedGas().SetWithLabel(int64(newBlock.Header().GasUsed()), map[string]string{"type": "received"})
 		metricBlockProcessedDuration().Observe(time.Duration(realElapsed).Milliseconds())
+		go metricsWriteTxData(n.forkConfig, newBlock)
```

Comment:
Following previous implementations of go routines for metrics, I think it would make sense to implement something similar to `EnableMetrics` https://github.com/vechain/thor/blob/release/galactica/muxdb/muxdb.go#L147 where we shut down the go routine when closing the DB https://github.com/vechain/thor/blob/release/galactica/muxdb/muxdb.go#L147

Reply:
Another option could be waiting groups, like in subscriptions https://github.com/vechain/thor/blob/release/galactica/api/subscriptions/subscriptions.go#L81 and closing https://github.com/vechain/thor/blob/release/galactica/api/subscriptions/subscriptions.go#L354